### PR TITLE
refactor: import absolute paths and sortings

### PR DESCRIPTION
**Pull request by enesozturk**
# Description

This PR introduces a refactor to our import statements across the project. By transitioning to **absolute paths**, we streamline module resolution and improve the maintainability of our codebase. This change ensures that our imports are clear, consistent, and less prone to errors caused by complex relative path navigation.

Additionally, I've implemented linting enhancements that enforce a standard for **import sorting** and organization. This doesn't just lead to cleaner code; it also assists in reducing merge conflicts related to import lines and facilitates quicker code reviews.

Key improvements include:

- Adoption of absolute paths for all internal module imports for better readability and maintenance
- Implementation of linting rules for consistent import order and grouping.

# Type of change

- [x] DX improvements (non-breaking change which improves developer experience and code quality)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,7 +3,6 @@
   "private": true,
   "version": "0.0.0",
   "license": "Apache-2.0",
-  "type": "module",
```
Comment (devceline) [inline package.json:6]: Why remove `type`

**Review by devceline — COMMENTED** [review]
This a great change, tired of dealing with imports when rebasing :D

File: package.json (hunk 1 above)
Comment (enesozturk) [reply package.json]: Due to issues in Prettier with ESModules, it cannot used with `.js` even though I tried `v3`. But in our case we don't need `.js` actually, so reverted back to `module` and upgraded Prettier now

**Comment by enesozturk** [comment]
> This a great change, tired of dealing with imports when rebasing :D

Exactly, huge burden

**Pull request merged**